### PR TITLE
add mesh lighting controls

### PR DESCRIPTION
**Pull request by kevinyamauchi**
This PR adds mesh lighting controls

### CODE REVIEW

**Comment by kevinyamauchi** [comment]
Example of setting the light direction to follow the camera:

https://user-images.githubusercontent.com/1120672/154949532-d7af239d-8c01-4038-a2ca-ac380e6267b4.mov

**Comment by alisterburt** [comment]
Awesome man, this looks really great! Trying to finish up the API stuff now so we can get both in :)

**Comment by kevinyamauchi** [comment]
@alisterburt , this is ready for review after #31 is merged

File: examples/mesh_headlight.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+# add the mri
+viewer.add_surface((brain_vertices, brain_faces, brain_vertex_depth), name='base')
+viewer.add_surface((brain_vertices, brain_faces, timeseries),
+                    colormap='turbo', opacity=0.9,
+                    contrast_limits=[-1.5, 3.5], name='timeseries')
+
+lighting_control_widget = LightingControlWidget(viewer=viewer)
+viewer.window.add_dock_widget(widget=lighting_control_widget)
+
+napari.run()
```
Comment (alisterburt) [inline examples/mesh_headlight.py:33]:
```suggestion
napari.run()

```

File: src/napari_threedee/lighting/lighting_control.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        return self._selected_layer_visuals
+
+    def _on_camera_change(self, event=None):
+        if self.enabled is False:
+            # only update lighting direction if enabled
+            return
+        view_direction = np.asarray(self._viewer.camera.view_direction)
+
+        for visual in self.selected_layer_visuals:
+            visual.node.shading_filter.light_dir = view_direction[::-1]
```
Comment (alisterburt) [inline src/napari_threedee/lighting/lighting_control.py:55]: should we add the transforms here too?

something like (untested)

```python
for layer, visual in zip(self._selected_layers, self._selected_layer_visuals):
    layer_view_direction = layer._world_to_data_ray(view_direction)
    visual.node.shading_filter.light_dir = layer_view_direction[::-1]
```

**Review by alisterburt — APPROVED** [review]
Great job man, had a quick play, the feature is awesome - minor suggestion about taking into account transforms before we forget but again, happy to merge as is!

File: src/napari_threedee/lighting/lighting_control.py (hunk 2 above)
Comment (kevinyamauchi) [reply src/napari_threedee/lighting/lighting_control.py:55]: Yes, that makes total sense. I'll do that before merging. Otherwise, I will forget!

Comment (kevinyamauchi) [reply src/napari_threedee/lighting/lighting_control.py]: Fixed!

**Comment by kevinyamauchi** [comment]
I fixed the transform issue and exposed the widget as a plugin. We are now up to 5 plugins!

**Pull request merged**